packages/calling/src/CallingClient/line/line.test.ts

```diff
@@ -0,0 +1,185 @@
+
+      expect(createRegistrationSpy).toBeCalledWith(
+        webex,
+        defaultServiceData,
+        expect.any(Mutex),
+        expect.anything(),
+        LOGGER.INFO
+      );
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.DEFAULT);
+      await line.register();
+
+      expect(webex.request).toBeCalledTimes(1);
+      expect(handleErrorSpy).not.toBeCalled();
+
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+      expect(line.getActiveMobiusUrl()).toEqual(primaryUrl);
+      expect(line.getLoggingLevel()).toEqual(LOGGER.INFO);
+      expect(line.getDeviceId()).toEqual(mockRegistrationBody.device.deviceId);
+
+      /** trigger sendKeepAlive and verify whether keepalive request is sent or not */
+      webex.request.mockClear();
+
+      /* specify keepalive interval as 30 seconds and advance timers by 30 */
+      line.sendKeepAlive({...mockRegistrationBody, keepaliveInterval: 30});
+
+      jest.advanceTimersByTime(30 * 1000);
+      await Promise.resolve();
+
+      expect(webex.request).toBeCalledWith({
+        ...getMockRequestTemplate(),
+        uri: `${mockRegistrationBody.device.uri}/status`,
+        method: 'POST',
+      });
+      jest.useRealTimers();
+    });
+
+    it('verify post registration api call when default mobius url is used', async () => {
+      const uri = `${webex.internal.services._serviceUrls.mobius}${URL_ENDPOINT}`;
+      webex.request.mockReturnValue(registrationPayload);
+      primaryMobiusUris.mockReturnValueOnce([uri]);
+
+      // calling it here as we need a different mobius uri to be passed in to constructor
+      line = new Line(
+        userId,
+        clientDeviceUri,
+        LineStatus.ACTIVE,
+        mutex,
+        primaryMobiusUris(),
+        backupMobiusUris(),
+        LOGGER.WARN
+      );
+
+      expect(createRegistrationSpy).toBeCalledWith(
+        webex,
+        defaultServiceData,
+        expect.any(Mutex),
+        expect.anything(),
+        LOGGER.WARN
+      );
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.DEFAULT);
+      await line.register();
+
+      expect(webex.request).toBeCalledTimes(1);
+      expect(handleErrorSpy).not.toBeCalled();
+
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+
+      expect(line.getActiveMobiusUrl()).toBe(uri);
+
+      expect(webex.request).toBeCalledWith({
+        method: 'POST',
+        ...getMockRequestTemplate(),
+        uri: `${uri}device`,
+        body: mockBody,
+      });
+    });
+
+    it('verify failure Registration cases all requests fail ', async () => {
+      webex.request.mockImplementation(() => {
+        throw new Error();
+      });
+
+      expect(line.getRegistrationStatus()).toBe(MobiusStatus.DEFAULT);
+      line.register();
+      await utils.waitForMsecs(20);
+
+      expect(line.getRegistrationStatus()).toBe(MobiusStatus.DEFAULT);
+      expect(handleErrorSpy).toBeCalled();
```

Comment:
When errorSpy is called we should compare what it is being called with. Here as well use toBeCalledOnceWith()

Reply:
For this line, here it's been called 8 times so we can't use `toBeCalledOnceWith` or `toBeCalledWith`. At other places where it's called only once, we have used `calledWith`. I have added `toBeCalledTimes(8)` at above line.

Reply:
💡 Tip: we can always use `mock.calls` or `nthCalledWith` in such scenarios. 

However, not needed in this case.
Although I am wondering how come there are 8 such invocations.

Reply:
Not sure how other test was affecting number of mock calls in this test as we're clearing all mocks after each test but after removing the default mobius url test case now this is called only once and I have added calledOnceWith here.